src/app/pcasts/dao/listening_histories_dao.py

```diff
@@ -1,25 +1,38 @@
 from app.pcasts.dao import episodes_dao
 from . import *
 
-def create_or_update_listening_history(episode_id, user):
-  optional_episode = episodes_dao.get_episode(episode_id, user.id)
-  if optional_episode:
-    optional_listening_history = ListeningHistory.\
-      query.\
-      filter(
-          ListeningHistory.episode_id == episode_id,
-          ListeningHistory.user_id == user.id
-      ).first()
-
-    if optional_listening_history:
-      optional_listening_history.update_listening_time()
-      db_utils.db_session_commit()
-      return optional_listening_history
-
-    listening_history = ListeningHistory(episode_id=episode_id, user_id=user.id)
-    return db_utils.commit_model(listening_history)
-  else:
-    raise Exception('Invalid episode_id provided')
+def create_or_update_listening_histories(episode_update_info_map, user):
+  episode_ids = episode_update_info_map.keys()
+  episodes = episodes_dao.get_episodes(episode_ids, user.id)
```

Comment:
why would you need to pass in `user.id` into `episodes`?

Reply:
client wants us to give them a fields like `is_bookmarked` and `is_recommended` from the perpsective of whoever is making the request for convenience purposes

Reply:
Is this necessary for the listening_histories?

Reply:
dont think so, maybe we can make that parameter optional and not attach those fields if nothing is passed in? it would make it slightly more efficient, but we should prob do that in another pr

Reply:
since we currently return the list of affected `ListeningHistory` objects, I think they do need the bookmarked and recommended field in case client wants to do anything with the resulting episodes. If they don't use the result, then we could return nothing and not do the computation to tack that info onto the episodes.